src/main/java/org/aksw/iguana/commons/io/ByteArrayListInputStream.java

```diff
@@ -125,12 +141,16 @@ public void skipNBytes(long n) throws IOException {
     }
 
     @Override
-    public int available() throws IOException {
+    public int available() {
         return (int) Math.min(Integer.MAX_VALUE, availableLong());
```

Comment:
```suggestion
        return (int) Math.min(Integer.MAX_VALUE - 8, availableLong());
```

Would probably be good to introduce a constant `MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8` for that